src/NHibernate.Test/UtilityTest/LinkHashMapFixture.cs

```diff
@@ -0,0 +1,474 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using NHibernate.Util;
+using NUnit.Framework;
+
+namespace NHibernate.Test.UtilityTest
+{
+	[TestFixture]
+	public class LinkHashMapFixture
+	{
+		private static readonly Player[] players =
+		{
+			new Player("12341", "Boeta Dippenaar"),
+			new Player("23432", "Gary Kirsten"),
+			new Player("23411", "Graeme Smith"),
+			new Player("55221", "Jonty Rhodes"),
+			new Player("61234", "Monde Zondeki"),
+			new Player("23415", "Paul Adams")
+		};
+
+		private static void Fill(IDictionary<string, Player> lhm)
+		{
+			foreach (var player in players)
+				lhm.Add(player.Id, player);
+		}
+
+		[Test]
+		public void Add()
+		{
+			IDictionary<string, Player> lhm = new LinkHashMap<string, Player>();
+			Fill(lhm);
+			lhm.Add("55555", new Player("55555", "Monde Zondeki"));
+
+			Assert.That(lhm.Count, Is.EqualTo(7));
+		}
+
+		[Test]
+		public void LastKeyLastValue()
+		{
+			var lhm = new LinkHashMap<string, Player>();
+			Fill(lhm);
+			Assert.That(lhm.LastKey, Is.EqualTo(players[players.Length - 1].Id));
+			Assert.That(lhm.LastValue, Is.EqualTo(players[players.Length - 1]));
+
+			// override
+			var antWithSameId = new Player("12341", "Another");
+			lhm[antWithSameId.Id] = antWithSameId;
+			Assert.That(lhm.LastKey, Is.EqualTo(antWithSameId.Id));
+			Assert.That(lhm.LastValue, Is.EqualTo(antWithSameId));
+		}
+
+		[Test]
+		public void FirstKeyFirstValue()
+		{
+			var lhm = new LinkHashMap<string, Player>();
+			Fill(lhm);
+			Assert.That(lhm.FirstKey, Is.EqualTo(players[0].Id));
+			Assert.That(lhm.FirstValue, Is.EqualTo(players[0]));
+
+			// override First
+			var antWithSameId = new Player("12341", "Another");
+			lhm[antWithSameId.Id] = antWithSameId;
+			Assert.That(lhm.FirstKey, Is.EqualTo(players[1].Id));
+			Assert.That(lhm.FirstValue, Is.EqualTo(players[1]));
+		}
+
+		[Test]
+		public void Clear()
+		{
+			IDictionary<string, Player> lhm = new LinkHashMap<string, Player>();
+			var p = new Player("78945", "Someone");
+			lhm[p.Id] = p;
+
+			lhm.Clear();
+			Assert.That(lhm, Is.Empty);
+
+			foreach (KeyValuePair<string, Player> pair in lhm)
+				Assert.Fail("Should not be any entries but found Key = " + pair.Key + " and Value = " + pair.Value);
+		}
+
+		[Test]
+		public void Contains()
+		{
+			var lhm = new LinkHashMap<string, Player>();
+			Fill(lhm);
+
+			Assert.That(lhm.Contains("12341"), Is.True);
+			Assert.That(lhm.Contains("55555"), Is.False);
```

Comment:
I do not change that because:
- The suggestion does not work, because `Does.Contain` checks existence against KeyValuePairs collection.
- `LinkHashMap.Contains(string)` is a target of this test.

Reply:
Ok.